src/main/java/io/a2a/spec/JSONRPCError.java

```diff
@@ -5,10 +5,44 @@
 /**
  * Represents a JSONRPC error.
  */
-public record JSONRPCError(int code, String message, Object data) {
+public class JSONRPCError {
```

Comment:
Just to check, any reason to convert this to a class? This is immutable and we're using Jackson to convert to/from JSON. It looks like the MCP Java SDK used records for errors like this.

Reply:
Ah, I see now, because the other errors need to extend this one, right?

Reply:
@fjuma Yeah, I wasn't able to inherit from Records. It says they are final